odnoklassniki-android-sdk/src/main/java/ru/ok/android/sdk/Odnoklassniki.java

```diff
@@ -409,10 +437,41 @@ private void signParameters(final Map<String, String> params) {
         params.put(Shared.PARAM_SIGN, sig);
     }
 
-    public final void setOkListener(OkListener listener) {
+    public final void setOkListener(final OkListener listener) {
+        sendDeferredData(listener);
+
         this.mOkListener = listener;
     }
 
+    /**
+     *  Sends deferred data to listeners
+     *
+     * @param listener
+     */
+    private void sendDeferredData(final OkListener listener) {
+        if (deferredDataList.size() > 0) {
```

Comment:
using ConcurrentLinkedQueue or analogues will speedup and simplify the corresponding code